doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,246 @@
+	      "state": "fail",
+	      "suggestions": [
+		"Add a spec descriptor for size"
+	      ],
+	      "crname": "example-memcached"
+      } 
+```
+
+The test image will be executed as a Pod by the scorecard with a restart policy of `never`.
+
+Also, for each test execution, scorecard would also make the
+bundle image contents available to test images by mounting
+the bundle image contents into the following location inside the test pod:
+
+```
+/scorecard/bundle/
+```
+
+This allows tests to have access to all the bundle contents if they 
+are required for the test logic.  The scorecard does not make assumptions about what the test does with this proposed design.  The test could for example:
+
+ * not use the CR's at all (e.g. doing a static bundle validation)
+ * use the CRs in the CSVs alm-examples annotation (e.g. to verify they validate the associated CRD schema)
+ * use some other CRs built (or maybe mounted) into the test image 
+
+For the case of executing a test, the bundle image will be passed as a scorecard command flag:
+```
+operator-sdk scorecard --bundle='quay.io/myorg/mytest' --selector=suite=basic
+```
+
+#### Test Output
+
+Custom tests are expected to produce log output that conforms to the
+scorecard v1alpha2 Test Result output.  This allows scorecard to parse
+the custom test log output and aggregate the results like any other
+scorecard test.
+
+The scorecard expected output would be in JSON, and would match the following definition:  https://github.com/operator-framework/operator-sdk/blob/master/pkg/apis/scorecard/v1alpha2/types.go#L36
+
+The scorecard v1alpha2 Test Result output contains information such as
+pass or fail status and suggestions/errors produced by the test that
+the end user would want to see in the aggregated scorecard results.
+
+
+### Custom Test Development Model
+
+The developer of a custom test would have a workflow similar to:
+ 
+ * write a custom test binary in the language of their choice, sample custom test within the SDK repository would be available as a starting point
+ * test the custom test binary locally (out-of-cluster) to make sure the output is produced that matches v1alpha2 ScorecardTestResult format
+ * scorecard creates a scorecard DSL config file for their test image
+ * when the test binary works as expected, the developer would build and push the test image to a container registry accessible to the scorecard test environment
+ * scorecard could build a bundle image that contains the scorecard DSL config
+ NOTE:  the Basic and OLM test manifests would be added automatically to the bundle image as part of the bundle image build process.  This allows users to run the Basic and OLM tests along with their custom tests.
+ * the developer would run the scorecard passing in their operator bundle image they just built along with any other scorecard flags (e.g. selector).
+
+The case of a developer implementing a kuttl test would be as follows:
```

Comment:
the default kuttl, and olm test images must be overridable IMO.

Reply:
I added a sample of the DSL that shows how default images could be overridden.